src/main/java/de/torui/coflsky/EventHandler.java

```diff
@@ -188,4 +202,165 @@ private static void ProcessScoreboard(String line){
+                    } else {
+                        tl.appendTag(EMPTY_COMPOUND);
+                    }
+                }
+
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                try {
+                    compound.setTag("i", tl);
+                    CompressedStreamTools.writeCompressed(compound, baos);
+
+                    wrapper.fullInventoryNbt = Base64.getEncoder().encodeToString(baos.toByteArray());
+
+                    String data = WSClient.gson.toJson(wrapper);
+                    String info = QueryServerCommands.PostRequest("https://sky.coflnet.com/api/mod/description/modifications", data);
+
+                    DescModification[][] arr = WSClient.gson.fromJson(info, DescModification[][].class);
+                    int i = 0;
+                    for (Slot obj : gc.inventorySlots.inventorySlots) {
+                        ItemStack stack = obj.getStack();
+                        if (stack != null) {
+                            tooltipItemMap.put(stack, arr[i]);
+                            String uuid = ExtractUuidFromItemStack(stack);
+                            if(uuid.length()>0){
+                                tooltipItemUuidMap.put(uuid, arr[i]);
+                            }
+
+                            String id = ExtractStackableIdFromItemStack(stack);
+                            if(id.length()>0){
+                                tooltipItemIdMap.put(id, arr[i]);
+                            }
+
+                        }
+                        i++;
+                    }
+
+                } catch (IOException e) {
+                }
+
+            }).start();
+        }
+    }
+
+    private static DescModification[] getTooltipData(ItemStack itemStack) {
+        if (tooltipItemMap.containsKey(itemStack)) {
+            return tooltipItemMap.getOrDefault(itemStack, EMPTY_ARRAY);
+        }
+        if(!itemStack.isStackable()){
+            String id = ExtractUuidFromItemStack(itemStack);
+            if (tooltipItemUuidMap.containsKey(id)) {
+                return tooltipItemUuidMap.getOrDefault(id, EMPTY_ARRAY);
+            }
+        } else {
+            String itemId = ExtractStackableIdFromItemStack(itemStack);
+            if(tooltipItemIdMap.containsKey(itemId)){
+                return tooltipItemIdMap.getOrDefault(itemId, EMPTY_ARRAY);
+            }
+        }
+
+        return EMPTY_ARRAY;
+    }
+
+    public static void onToolTipEventHandler(ItemTooltipEvent event) {
+        DescModification[] data = getTooltipData(event.itemStack);
+
+        if (data == null || data.length == 0)
+            return;
+
+        for (int i = 0; i < data.length; i++) {
+            if (data[i].type.equals("APPEND")){
+                event.toolTip.add(data[i].value);
+            } else if (data[i].type.equals("REPLACE")) {
+                try {
+                    for (int i2 = 0; i2 < event.toolTip.size(); i2++) {
```

Comment:
this make can be optimized, the REPLACE originally meant to replace text in the whole lore.  
I reviewed the Java documentation and am now sending `line` instead of `target` to replace a specific line.  
Please update this function accordingly